src/web/wms/src/test/java/org/geoserver/wms/web/data/StyleEditPageChangeWorkspaceTest.java

```diff
@@ -0,0 +1,94 @@
+/* (c) 2017 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms.web.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.FileReader;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.wicket.markup.html.form.DropDownChoice;
+import org.apache.wicket.util.file.File;
+import org.apache.wicket.util.tester.FormTester;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.StyleInfo;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.web.GeoServerWicketTestSupport;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests for moving styles between workspaces from the {@link StyleEditPage}.
+ */
+public class StyleEditPageChangeWorkspaceTest extends GeoServerWicketTestSupport {
+
+    private static final String TEST_STYLE_NAME = "testStyle";
+
+    private static final String TEST_STYLE_FILENAME = "testMoveStyle.sld";
+
+    private static final String TEST_STYLE_FILENAME_UPDATED = "testMoveStyleUpdated.sld";
+
+    StyleInfo styleInfo;
+
+    StyleEditPage edit;
+
+    @Before
+    public void setUp() throws Exception {
+        Catalog catalog = getCatalog();
+        login();
+        styleInfo = catalog.getStyleByName("testStyle");
+        edit = new StyleEditPage(styleInfo);
+        tester.startPage(edit);
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        testData.addStyle(TEST_STYLE_NAME, TEST_STYLE_FILENAME, this.getClass(), getCatalog());
+    }
+
+    /**
+     * Test that a user can update the .sld file contents and move the style into a workspace in a single edit.
+     */
+    @Test
+    public void testMoveWorkspaceAndEdit() throws Exception {
```

Comment:
Given that this is just a single test, it might be better of in StyleEditPageTest.java rather than a separate file. Its probably fine as-is though